cypress/integration/save.cy.js

```diff
@@ -0,0 +1,182 @@
+    selectRelativePeriods,
+} from '../elements/dimensionModal'
+import { changeVisType } from '../elements/visualizationTypeSelector'
+import {
+    expectAOTitleToBeDirty,
+    expectAOTitleToBeUnsaved,
+    expectAOTitleToBeValue,
+    expectAOTitleToNotBeDirty,
+    expectVisualizationToBeVisible,
+} from '../elements/chart'
+import { TEST_DATA_ELEMENTS } from '../utils/data'
+import {
+    //openRandomAOCreatedByOthers,
+    saveNewAO,
+    closeFileMenu,
+    saveAOAs,
+    saveExistingAO,
+    openAOByName,
+    deleteAO,
+    expectFileMenuButtonToBeEnabled,
+    expectFileMenuButtonToBeDisabled,
+    FILE_MENU_BUTTON_SAVEAS,
+    FILE_MENU_BUTTON_DELETE,
+    FILE_MENU_BUTTON_RENAME,
+    FILE_MENU_BUTTON_TRANSLATE,
+    FILE_MENU_BUTTON_SHARE,
+    FILE_MENU_BUTTON_GETLINK,
+} from '../elements/fileMenu'
+import { expectRouteToBeAOId, expectRouteToBeEmpty } from '../elements/route'
+import { getRandomVisType } from '../utils/random'
+import {
+    clickMenuBarFileButton,
+    clickMenuBarUpdateButton,
+} from '../elements/menuBar'
+import { selectYoyCategoryOption } from '../elements/layout'
+import {
+    expectStartScreenToBeVisible,
+    goToStartPage,
+} from '../elements/startScreen'
+
+const TEST_VIS_NAME = `TEST ${new Date().toLocaleString()}`
+const TEST_VIS_NAME_UPDATED = `${TEST_VIS_NAME} - updated`
+const TEST_VIS_DESCRIPTION = 'Generated by Cypress'
+const TEST_VIS_TYPE = getRandomVisType()
+const TEST_VIS_TYPE_NAME = visTypeDisplayNames[TEST_VIS_TYPE]
+
+// TODO: Add test to check that the description is saved and shown in the interpretations panel
+
+describe('saving an AO', () => {
+    describe('"save" and "save as" for a new AO', () => {
+        it('navigates to the start page', () => {
+            goToStartPage()
+        })
+        it(`changes vis type to ${TEST_VIS_TYPE_NAME}`, () => {
+            changeVisType(TEST_VIS_TYPE_NAME)
+        })
+        it('adds Data dimension items', () => {
+            openDimension(DIMENSION_ID_DATA)
+            selectDataElements(
+                TEST_DATA_ELEMENTS.slice(1, 2).map(item => item.name)
+            )
+            clickDimensionModalUpdateButton()
+        })
+        it('displays an unsaved visualization', () => {
+            expectVisualizationToBeVisible(TEST_VIS_TYPE)
+            expectAOTitleToBeUnsaved()
+            expectRouteToBeEmpty()
+        })
+        it('checks that Save as is disabled', () => {
+            clickMenuBarFileButton()
+            expectFileMenuButtonToBeDisabled(FILE_MENU_BUTTON_SAVEAS)
+            closeFileMenu()
+        })
+        it('saves new AO using "Save"', () => {
+            saveNewAO(TEST_VIS_NAME, TEST_VIS_DESCRIPTION)
+            expectAOTitleToBeValue(TEST_VIS_NAME)
+            expectVisualizationToBeVisible(TEST_VIS_TYPE)
+        })
+        it('checks that the url was changed', () => {
+            expectRouteToBeAOId()
+        })
+        it('all File menu buttons are enabled', () => {
```

Comment:
This doesn't actually check all the File menu buttons? (NEW, SAVE, OPEN missing)